### DISCUSSION TITLE
Optimize Level Icons by cropping them and resizing them when they get uploaded

### DISCUSSION THREAD

**Discussion by FuchsiaTheJellyfish** [LBPUnion/ProjectLighthouse#1038] [Ideas]
I've noticed a little quirk in the beacon website. The level icons do not appear to be resized or compressed, which is not good for optimization. I've made this discussion just to let you know about this issue.

Oh and resize the beacon icon on the top menu it does NOT need to be 2560x2560 please

Edit: You should try going for a low size, maybe 512x512, or 256x256? As for the icon I recommend you make it 128x128 and use a different size for the bigger icons, maybe 256x256 or 512x512.

I think it'll help your website load faster.

I probably sound like I'm asking for too much, but this is coming from someone who isn't a big fan of unoptimized atrocities like the 2560x2560 icon.

**Comment by m88youngling** [comment]
I have also noticed this. My question is, how difficult would this be to fix vs putting it on the agenda for Lighthouse 2?

**Reply by FuchsiaTheJellyfish** [reply]
For the icon's atrocious size, you should try resizing it. Pick a small size that i've suggested for you. I'm not too sure about the cropping and resizing part, as I'm still learning IT, but if Refresh can do it, then it's possible.

**Comment by TorutheRedFox** [comment]
> Oh and resize the beacon icon on the top menu it does NOT need to be 2560x2560 please

okay but why is it not an SVG if it needs that much detail

**Reply by m88youngling** [reply]
Because I didn't know how to make SVGs 💀

**Reply by jackcaver** [reply]
The funny thing is, the repo DOES HAVE an SVG version of the icon...

**Reply by jackcaver** [reply]
Tbf 2000x2000 is already too much, but for some reason the pride version is 2560x2560 O-o

**Comment by vilijur** [comment]
tbh not sure why this hasn't been implemented yet

not only would it be good for optimisation, it would also prevent lbp from crashing if an image for a level or profile picture is too big
also having it be cropped would be good for preventing lbp1 from squishing the icons

**Comment by m88youngling** [comment]
Could we make it so that there's the default logo, but you can customize it with lighthouse.yml to have a custom logo similiar to what Infinite does? I'm not sure how they implemented that in their fork

**Reply by jackcaver** [reply]
you just need to replace files that have the logo to do that...

**Reply by TorutheRedFox** [reply]
iirc the logo is stored externally so you don't even need to recompile it to do that

**Reply by sudokoko** [reply]
it's stored in the StaticFiles directory that ASP.NET generates

**Reply by m88youngling** [reply]
I guess this is on my to do list then. I should probably also make a PR with the corrected size so that new instances can have the correct size to start with
